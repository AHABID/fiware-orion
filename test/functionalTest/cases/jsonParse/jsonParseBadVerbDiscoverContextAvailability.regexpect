+++++ Starting +++++++++++++++
HTTP/1.1 200 OK
Content-Length: 75
Content-Type: application/json
Date: REGEX(.*)

{
    "duration": "PT1M",
    "registrationId": "REGEX([0-9a-f]{24})"
}
+++++ Registration Done +++++++++++++++
HTTP/1.1 405 Method Not Allowed
Content-Length: 0
Allow: POST
Date: REGEX(.*)

+++++ Discovery Done +++++++++++++++
