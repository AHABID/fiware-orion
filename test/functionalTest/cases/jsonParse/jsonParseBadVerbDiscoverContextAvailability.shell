echo "+++++ Starting +++++++++++++++"
url="/NGSI9/registerContext"
payload='{
  "contextRegistrations": [
    {
      "entities": [
        {
          "type": "Room",
          "isPattern": "false",
          "id": "ConferenceRoom"
        },
        {
          "type": "Room",
          "isPattern": "false",
          "id": "OfficeRoom"
        }
      ],
      "attributes": [
        {
          "name": "temperature",
          "type": "degree",
          "isDomain": "false",
          "metadatas": [
            {
              "name": "ID",
              "type": "string",
              "value": "1110"
            }
          ]
        }
      ],
      "metadatas": [
        {
          "name": "ID",
          "type": "string",
          "value": "2212"
        }
      ],
      "providingApplication": "http://192.168.100.1:70/application"
    }
  ],
  "duration": "PT1M"
}'
curlJson "$url" "$payload"

echo "+++++ Registration Done +++++++++++++++"

url="/NGSI9/discoverContextAvailability"
payload='{
  "entities": [
    {
      "type": "Room",
      "isPattern": "false",
      "id": "ConferenceRoom"
    },
    {
      "type": "Room",
      "isPattern": "false",
      "id": "OfficeRoom"
    }
  ],
  "attributes": [
    "temperature",
    "pressure"
  ],
  "restriction": {
    "attributeExpression" : "Attribute Expression",
    "scopes": [
      {
        "type" : "st1",
        "value" : "sv1"
      },
      {
        "type" : "st2",
        "value" : "sv2"
      }
    ]
  }
}'
curlJson "$url" "$payload" "-X BAD_VERB"

echo "+++++ Discovery Done +++++++++++++++"
