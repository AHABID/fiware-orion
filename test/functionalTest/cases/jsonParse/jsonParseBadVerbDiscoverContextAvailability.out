+++++ Starting +++++++++++++++
HTTP/1.1 200 OK
Content-Length: 75
Content-Type: application/json
Date: Tue, 03 Jun 2014 09:46:06 GMT

{
    "duration": "PT1M", 
    "registrationId": "538d995e813cbf3d070ff2d4"
}
+++++ Registration Done +++++++++++++++
HTTP/1.1 405 Method Not Allowed
Content-Length: 0
Allow: POST
Date: Tue, 03 Jun 2014 09:46:06 GMT

+++++ Discovery Done +++++++++++++++
